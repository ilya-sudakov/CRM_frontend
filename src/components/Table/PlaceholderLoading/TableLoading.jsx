import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import './TableLoading.scss';

const TableLoading = ({
  items = 3,
  columnLength = 3,
  ItemElement,
  WrapperElement,
}) => {
  const [elements, setElements] = useState([]);

  useEffect(() => {
    let temp = [];
    const count = items;
    for (let i = 1; i <= count; i++) {
      temp.push(
        <WrapperElement>
          {new Array(columnLength).fill(0).map((_, index) => (
            <ItemElement key={index}>
              <div className="table-loading__item"></div>
            </ItemElement>
          ))}
        </WrapperElement>,
      );
    }
    setElements([...temp]);
  }, []);

  return elements.map((item) => item);
};

export default TableLoading;

TableLoading.propTypes = {
  items: PropTypes.number,
  minHeight: PropTypes.string,
  itemClassName: PropTypes.string,
  wrapperClassName: PropTypes.string,
  placeholderContent: PropTypes.any,
};
