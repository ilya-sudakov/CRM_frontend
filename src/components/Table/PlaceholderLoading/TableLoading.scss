.table-loading__item {
    @keyframes pulse-placeholder-loading {
        0%,
        100% {
            background-color: #e7e7e7;
        }
        50% {
            background-color: #efefef;
        }
    }

    @keyframes load {
        from {
            left: -150px;
        }
        to {
            left: 100%;
        }
    }
    height: 1rem;
    background-color: #ddd;
    animation: pulse-placeholder-loading 1s infinite ease-in-out;
    position: relative;
    overflow: hidden;
    border-bottom: 1px solid #bbb;

    &::after {
        content: "";
        display: block;
        position: absolute;
        left: -150px;
        top: 0;
        height: 100%;
        width: 150px;
        background: linear-gradient(to right, transparent 0%, #e8e8e8 50%, transparent 100%);
        animation: load 1s cubic-bezier(0.4, 0, 0.2, 1) infinite;
    }
}
