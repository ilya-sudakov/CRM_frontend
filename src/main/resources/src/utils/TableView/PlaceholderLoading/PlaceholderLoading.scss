.placeholder-loading {
    @keyframes pulse1 {
        0% {
            border-color: #dbdbdb;
            background-color: #dbdbdb;
        }
        50% {
            border-color: #c5c5c5;
            background-color: #c5c5c5;
        }
        100% {
            border-color: #dbdbdb;
            background-color: #dbdbdb;
        }
    }

    @keyframes load {
        from {
            // background: linear-gradient(to right, transparent 0%, #fafafa 50%, transparent 100%);
            left: -150px;
        }
        to {
            // background: linear-gradient(to right, transparent 0%, #e8e8e8 50%, transparent 100%);
            left: 100%;
        }
    }

    &__item {
        height: 1rem;
        background-color: #dddddd;
        animation: pulse1 1s infinite ease-in-out;
        margin-bottom: 5px;
        border-radius: 5px;
        position: relative;
        overflow: hidden;
        // box-shadow: 0 4px 10px 0 rgba(33, 33, 33, 0.15);

        &::after {
            content: "";
            display: block;
            position: absolute;
            left: -150px;
            top: 0;
            height: 100%;
            width: 150px;
            background: linear-gradient(to right, transparent 0%, #e8e8e8 50%, transparent 100%);
            animation: load 1s cubic-bezier(0.4, 0, 0.2, 1) infinite;
        }
    }
}
