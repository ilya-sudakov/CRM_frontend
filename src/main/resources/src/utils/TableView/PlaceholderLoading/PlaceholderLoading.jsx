import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import "./PlaceholderLoading.scss";

const PlaceholderLoading = (props) => {
  const [elements, setElements] = useState([]);

  useEffect(() => {
    let temp = [];
    const count = props.items || 3;
    for (let i = 1; i <= count; i++) {
      temp.push(
        <div
          className={`placeholder-loading__item ${props.itemClassName}`}
          style={{ minHeight: `${props.minHeight || "1.5rem"}` }}
        ></div>
      );
    }
    setElements([...temp]);
  }, []);

  if (props.placeholderContent) {
    return (
      <div className={`placeholder-loading ${props.wrapperClassName}`}>
        {elements.map(() => (
          <div
            className={` ${props.itemClassName}`}
            style={{ minHeight: `${props.minHeight || "1.5rem"}` }}
          >
            {props.placeholderContent}
          </div>
        ))}
      </div>
    );
  }
  return (
    <div className={`placeholder-loading ${props.wrapperClassName}`}>
      {elements.map((item) => item)}
    </div>
  );
};

export default PlaceholderLoading;

PlaceholderLoading.propTypes = {
  items: PropTypes.number,
  minHeight: PropTypes.string,
  itemClassName: PropTypes.string,
  wrapperClassName: PropTypes.string,
  placeholderContent: PropTypes.any,
};
