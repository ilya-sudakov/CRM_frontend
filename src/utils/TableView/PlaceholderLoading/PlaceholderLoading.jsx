import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import './PlaceholderLoading.scss';

const PlaceholderLoading = ({
  items = 3,
  itemClassName,
  minHeight = '1.5rem',
  placeholderContent,
  wrapperClassName,
}) => {
  const [elements, setElements] = useState([]);

  useEffect(() => {
    let temp = [];
    const count = items;
    for (let i = 1; i <= count; i++) {
      temp.push(
        <div
          className={`placeholder-loading__item ${itemClassName}`}
          style={{ minHeight: minHeight }}
        ></div>,
      );
    }
    setElements([...temp]);
  }, []);

  if (placeholderContent) {
    return (
      <div className={`placeholder-loading ${wrapperClassName}`}>
        {elements.map(() => (
          <div className={itemClassName} style={{ minHeight: minHeight }}>
            {placeholderContent}
          </div>
        ))}
      </div>
    );
  }
  return (
    <div className={`placeholder-loading ${wrapperClassName}`}>
      {elements.map((item) => item)}
    </div>
  );
};

export default PlaceholderLoading;

PlaceholderLoading.propTypes = {
  items: PropTypes.number,
  minHeight: PropTypes.string,
  itemClassName: PropTypes.string,
  wrapperClassName: PropTypes.string,
  placeholderContent: PropTypes.any,
};
