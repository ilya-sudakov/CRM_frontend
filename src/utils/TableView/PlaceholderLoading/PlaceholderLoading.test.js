import React from 'react';
import PlaceholderLoading from './PlaceholderLoading.jsx';
import { render, cleanup, fireEvent, screen } from '@testing-library/react';

describe('PlaceholderLoading component', () => {
  afterEach(cleanup);

  it('renders w/o props', () => {
    render(<PlaceholderLoading />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<PlaceholderLoading />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders 20 items', () => {
    const { container } = render(
      <PlaceholderLoading items={20} itemClassName="test" />,
    );
    expect(container.getElementsByClassName('test').length).toBe(20);
  });

  it('renders custom content', () => {
    const { container } = render(
      <PlaceholderLoading
        items={5}
        itemClassName="test"
        placeholderContent={
          <div className="inner-test">
            <div className="inner-test"></div>
          </div>
        }
      />,
    );
    expect(container.getElementsByClassName('inner-test').length).toBe(10);
  });
});
