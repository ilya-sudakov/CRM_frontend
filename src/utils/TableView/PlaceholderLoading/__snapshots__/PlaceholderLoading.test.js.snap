// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderLoading component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="placeholder-loading undefined"
  >
    <div
      class="placeholder-loading__item undefined"
      style="min-height: 1.5rem;"
    />
    <div
      class="placeholder-loading__item undefined"
      style="min-height: 1.5rem;"
    />
    <div
      class="placeholder-loading__item undefined"
      style="min-height: 1.5rem;"
    />
  </div>
</DocumentFragment>
`;
